src/sonic-config-engine/minigraph.py

```diff
@@ -1714,7 +1765,14 @@ def get_tunnel_entries(tunnel_intfs, lo_intfs, hostname):
     for type, tunnel_dict in tunnel_intfs.items():
         for tunnel_key, tunnel_attr in tunnel_dict.items():
             tunnel_attr['dst_ip'] = lo_addr
+
+            if (tunnel_qos_remap.get('status') == 'enabled') and (mux_tunnel_name == tunnel_key) and (peer_switch_ip is not None):
+                tunnel_attr['src_ip'] = peer_switch_ip
```

Comment:
Since the minigraph is anyways changing, can we have the `src_ip` to be coming from minigraph `TunnelInterface `instead of having the logic to determine from  `peer_switch_ip` ?

Reply:
I was also thinking about that. The problem is, I'm afraid it's not easy to implement that logic in Metadata.
For example, for upper_tor, the peer_ip would be `10.1.0.33`, while for `lower_tor`, it would be `10.1.0.32`. That means we have to generate different minigraph for upper_tor/lower_tor. I don't konw if it's doable.

Reply:
IMO, this would be pretty simple to implement. And btw, the minigraph is indeed different for upper and lower tor. Approving this!